File: Unit4A5.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-25
 * @fileoverview This program will print out a times table for the values from 1 to 12.
 */

// variables
let row = "";

// output
console.log("Times Table from 1 to 12:");

// Loop through the numbers 1 to 12 for the rows
for (let rowVariable = 1; rowVariable <= 12; rowVariable++) {
  // Loop through the numbers 1 to 12 for the columns
  for (let columnVariable = 1; columnVariable <= 12; columnVariable++) {
    // Calculate the product and format it to align properly
    // .padStart(4, ' ') ensures each number takes up 4 characters for alignment
    row += (rowVariable * columnVariable).toString().padStart(4, ' ');
  }
  // Print the completed row
  console.log(row);
  // Reset the row variable for the next iteration
  row = "";
}

console.log("\nDone.");

File: Unit4A5.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-11-25
// Fileoverview: This program will print out a times table for the values from 1 to 12, for each of the 12 values in the table.

package main

import (
	"fmt"
)

func main() {
	// variables
	var row string

	// output
	fmt.Println("Times Table from 1 to 12:")

	// Loop through the numbers 1 to 12 for the rows
	for rowVariable := 1; rowVariable <= 12; rowVariable++ {
		// Loop through the numbers 1 to 12 for the columns
		for columnVariable := 1; columnVariable <= 12; columnVariable++ {
			// Calculate the product and format it to align properly
			// fmt.Sprintf("%4d", ...) formats the integer to take up 4 characters
			row += fmt.Sprintf("%4d", rowVariable*columnVariable)
		}
		// Print the completed row
		fmt.Println(row)
		// Reset the row variable for the next iteration
		row = ""
	}

	fmt.Println("\nDone.")
}
